### PR TITLE
Fix missing return statements on function literals

### PR DESCRIPTION
**Pull request by srawlins**
## Description

This PR fixes missing return statements that will be enforced in an upcoming build of Dart, when https://dart-review.googlesource.com/c/sdk/+/100301 lands.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by srawlins** [comment]
Merry Christmas, @goderbauer

File: packages/flutter_tools/lib/src/run_hot.dart [hunk 1]
```diff
@@ -444,7 +444,7 @@ class HotRunner extends ResidentRunner {
               final ServiceEvent pauseEvent = view.uiIsolate.pauseEvent;
               if ((pauseEvent != null) && pauseEvent.isPauseEvent) {
                 // Resume the isolate so that it can be killed by the embedder.
-                return view.uiIsolate.resume();
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/run_hot.dart]: This seems wrong

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/run_hot.dart]: Let me investigate what this code is supposed to do ...

**Comment by goderbauer** [comment]
> Merry Christmas, @goderbauer

It's the best time of the year!

**Review by goderbauer — APPROVED** [review]
This LGTM once @jonahwilliams is happy.

Thanks, @srawlins!

File: packages/flutter_tools/lib/src/run_hot.dart (hunk 1 above)
Comment (srawlins) [reply packages/flutter_tools/lib/src/run_hot.dart]: Yeah this one's super weird. AFAICT, there is no reason `then(THIS)` should have to return something. I think the error,

```
This function has a return type of 'Future<Map<String, dynamic>>', but doesn't
end with a return statement • packages/flutter_tools/lib/src/run_hot.dart:443:13
```

is saying that one path returns a `Future<Map<String, dynamic>>`, and another pass doesn't return anything, and those two paths then don't match any consistent return type. This return value isn't used anywhere, right?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/run_hot.dart]: The whenComplete triggers off of the returned future, or the immediate result of `return null`. in this case it should be:

```
if ((pauseEvent != null) && pauseEvent.isPauseEvent) {
  return view.uiIsolate.resume();
}
return null;
```
```

Comment (srawlins) [reply packages/flutter_tools/lib/src/run_hot.dart]: Done!

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by srawlins** [comment]
Er, I hope the windows `syntax_error_test.broken_dart` failures are unrelated?

**Comment by jonahwilliams** [comment]
I would pull upstream and re push

**Comment by srawlins** [comment]
Yay all passed, someone merge for me? 😄

**Pull request merged**